controller/controller_client.go

```diff
@@ -189,11 +197,20 @@ func (c *clientHandler) handleUnitAdded(unit *client.Unit) {
 
 // handle the UnitChangedModified event from the V2 API
 func (c *clientHandler) handleUnitUpdated(unit *client.Unit) {
-	state, logLvl, _ := unit.Expected()
-	c.log.Debugf("Got unit updated for ID %s (%s/%s)", unit.ID(), state.String(), logLvl.String())
+	state, logLvl, cfg := unit.Expected()
+	c.log.Infof("Got unit updated for ID %s (%s/%s)", unit.ID(), state.String(), logLvl.String())
+	currentUnit := c.units.GetInputUnit(unit.ID(), unit.Type())
+	// check to see if only the log level needs updating
+	if onlyLogLevelUpdated(cfg.Source.AsMap(), currentUnit.config, logLvl) {
```

Comment:
FYI we are planning to make this easier to detect https://github.com/elastic/elastic-agent-client/pull/56#issuecomment-1423056770